shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/repository/ResetPasswordRepositoryImpl.kt

```diff
@@ -0,0 +1,14 @@
+package co.nimblehq.blisskmmic.data.repository
+
+import co.nimblehq.blisskmmic.data.model.ResetPasswordMeta
+import co.nimblehq.blisskmmic.data.network.datasource.NetworkDataSource
+import co.nimblehq.blisskmmic.data.network.target.ResetPasswordTargetType
+import co.nimblehq.blisskmmic.domain.repository.ResetPasswordRepository
+import kotlinx.coroutines.flow.Flow
+
+class ResetPasswordRepositoryImpl(private val networkDataSource: NetworkDataSource): ResetPasswordRepository {
+
+    override fun reset(email: String): Flow<ResetPasswordMeta> {
+        return networkDataSource.resetPassword(ResetPasswordTargetType(email))
+    }
+}
```

Comment:
Having a second thoughts, I am wondering if we need to create a single-action repository file like this 🤔 because let's say we will need to support resetting email, username, etc. later on, we will have to create a new repository per functionality with this way. What do you think about grouping it as part of the `AuthenticationRepository` as it helps update the password for authentication features anyway?

Reply:
@minhnimble I think Authentication is for authentication, anything related to the token, and is having a lot of responsibility at this point.
I think reset password/email/username should be another repository. What do you think of the name `AccountRecoveryRepository`, in that case it can be all things related to getting the account working again?

Reply:
`AccountRecoveryRepository` sounds good to me also @blyscuit 👍